libs/exavmlib/mix.lock

```diff
@@ -1,7 +1,9 @@
 %{
```

Comment:
Don't we want to leave the lock file empty... otherwise won't these pinned versions will need to be updated each release?

Reply:
no we want reproducible builds, otherwise whatever scripting we do could break at anytime, due to bugs in new dep releases.

Reply:
Also especially as we are not stdlib elixir and a patch dep release may use Enum.xy that breaks due to the unique properties of exavmlib.

Reply:
Just so I understand, these pinned versions will work with any of the OTP/Elixir versions supported by AtomVM all the way back to OTP-21 with Elixir 1.7?

Reply:
I assume no, surely modern ex_doc uses newer goodies - but the mix.lock is only used for these docs purposes.

Reply:
Just for reference I was able to build the docs with mix using OTP-24 with Elixir-1.14.5. I was not able to build an OTP earlier than 24 on my test system because of openssl version incompatibilities (and I didn't want to download and build the older sources for OpenSSL from scratch). It seems ex_doc has a requirement for Elixir ~1.14, so that is the oldest version that can be used (which should work with OTP-23 - but I was unable to confirm)